python/ray/data/dataset.py

```diff
@@ -2591,6 +2574,9 @@ def make_generator():
                         f"but got a `{type(feature_columns).__name__}` instead."
                     )
 
+                if features.ndim == 1:
+                    features = tf.expand_dims(features, axis=1)
```

Comment:
Under what cases do we need this?

Reply:
+1

Reply:
We need it to not break existing tests and examples.

In the previous implementation, features would always get unsqueezed. I've changed the behavior such that features only get unsqueezed if the features have one dimension.

Reply:
Hmm shouldn't we only be unsqueezing if the features have **more** than one dimension? I'm of the opinion that we should always eliminate unit column dimensions.

Reply:
I'm not sure if I understand. Could you elaborate on when we'd unsqueeze with more than one dimension?

My understanding is that the only case you'd want to unsqueeze is when you have features of shape (B,) but your model expects shape (B, 1).

Reply:
Not sure if this is related, but columns get unsqueezed as a part of the `convert_pandas_to_tf_tensor` concatenation process. That's purely an implementation detail, though

Reply:
It comes down to what we'd gauge to be the most common use case: in the single feature case, is it more common for a user's model to expect a squeezed feature tensor, `(B, 32, 32)`, or an unsqueezed feature tensor, `(B, 1, 32, 32)`? If the answer isn't clear, then I'd vote for the default representation to be the least redundant, i.e. where the unit feature dimension is squeezed. This can also happen in another PR if we're trying to keep this PR focused.

I'm actually not sure if this check is right/desirable, since it looks like the behavior varies based on the column shape (scalar vs tensor columns). I.e., it would expand `(B,)` to `(B, 1)`, but not `(B, 32, 32)` to `(B, 1, 32, 32)`. 🤔

Reply:
> in the single feature case, is it more common for a user's model to expect a squeezed feature tensor, (B, 32, 32), or an unsqueezed feature tensor, (B, 1, 32, 32)?

My assumption is that the user's model expects an unsqueezed tensor in the `ndim == 1` case, and the unmodified tensor otherwise. I can't imagine any case where you're working with multidimensional data (e.g., images) and wanting an extra axis.

Reply:
Could we defer the topic of unsqueezing semantics to a different PR? I could just go with the implementation that minimizes breaking changes for now. I mostly just want to fix https://github.com/ray-project/ray/issues/25135 because I have to a bunch of jank workarounds in #24633 otherwise.

Reply:
Yep that sounds good!